crates/celo-alloy/consensus/src/receipt/envelope_alt.rs

```diff
@@ -0,0 +1,372 @@
+//! Receipt envelope types for Celo.
+
+use alloy_consensus::{Eip658Value, Receipt, ReceiptWithBloom, TxReceipt};
+use alloy_eips::{
+    Typed2718,
+    eip2718::{Decodable2718, Eip2718Error, Eip2718Result, Encodable2718, IsTyped2718},
+};
+use alloy_primitives::{Bloom, Log, logs_bloom};
+use alloy_rlp::{BufMut, Decodable, Encodable, length_of_length};
+use op_alloy_consensus::{OpDepositReceipt, OpDepositReceiptWithBloom, OpReceiptEnvelope};
+use std::vec::Vec;
+use crate::CeloTxTypeAlt;
+
+/// Receipt envelope, as defined in [EIP-2718], modified for Celo.
+///
+/// This enum distinguishes between tagged and untagged legacy receipts, as the
+/// in-protocol merkle tree may commit to EITHER 0-prefixed or raw. Therefore
+/// we must ensure that encoding returns the precise byte-array that was
+/// decoded, preserving the presence or absence of the `TransactionType` flag.
+///
+/// Transaction receipt payloads are specified in their respective EIPs.
+///
+/// [EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+/// 
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(tag = "type"))]
+pub enum CeloReceiptEnvelopeAlt<T = Log> {
+    /// OP envelope
+    OP(OpReceiptEnvelope<T>),
+    /// Receipt envelope with type flag 123, containing a [CIP-64] receipt.
+    ///
+    /// [CIP-64]: https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0064.md
+    #[cfg_attr(feature = "serde", serde(rename = "0x7b", alias = "0x7B"))]
+    Cip64(ReceiptWithBloom<Receipt<T>>), // TODO: replace with CeloCip64Receipt which includes baseFee
+}
```

Comment:
I just discovered https://docs.rs/alloy-consensus/latest/alloy_consensus/extended/enum.Extended.html, which seems like a good way extend the tx types. But it is new in alloy-consensus 1.0.7, so can't use it with our current version right away.